### PR TITLE
Move storage examples

### LINKED ISSUE #23671: Consolidate storage examples in to single dir
Consolidate all the storage examples (for example `examples/cephfs/`, `examples/iscsi/`, etc) in to a single directory under `examples/` (maybe `examples/storage`).

### PR DESCRIPTION
**Pull request by ciwang**
Fixes #23671 

Consolidate storage examples into separate folders for storage and volume plugins

### CODE REVIEW

**Comment by saad-ali** [comment]
@k8s-bot ok to test

File: CHANGELOG.md [hunk 1]
```diff
@@ -767,7 +767,7 @@ container in the pod will run with the FSGroup as a supplemental group
 Pod’s SELinux context, if specified
   * A stable client library release\_1\_2 is added. The library is [here](pkg/client/clientset_generated/release_1_2/), and detailed doc is [here](docs/devel/generating-clientset.md#released-clientsets). We will keep the interface of this go client stable.
   * New Azure File Service Volume Plugin enables mounting Microsoft Azure File
```
Comment (saad-ali) [inline CHANGELOG.md]: Don't update the Changelog

File: test/e2e/generated/bindata.go [hunk 2]
```diff
@@ -5,9 +5,9 @@
 // examples/aws_ebs/aws-ebs-web.yaml
```
Comment (saad-ali) [inline test/e2e/generated/bindata.go]: Do not manually edit generated files. The auto-generator should update this.

Comment (saad-ali) [inline test/e2e/generated/bindata.go]: Ditto for other auto-gen files in this commit. This commit should only contain manual edits. Read the comments at the top of the files to figure out if they should be manually edited or auto-generated.

**Comment by saad-ali** [comment]
The E2E failures looks like a flake: https://github.com/kubernetes/kubernetes/issues/26210

Looks like you have two real failures:
```
./hack/test-cmd.sh:568
./hack/verify-munge-docs.sh
```

See if you can fix those. If not, modify your 2nd commit (find and replace paths) to not touch auto-generated files. And replace your third and fourth commits with one commit that updates all the auto-generated files using `update-all` (make sure to run `make clean` first).

**Comment by ciwang** [comment]
It looks like since verify-munge-docs checks the links in the Changelog, it has to be updated in order for tests to pass. I'll push the version that passes test for now and can revert it if necessary.

**Comment by ciwang** [comment]
@k8s-bot node e2e test this issue: #27296

**Comment by ciwang** [comment]
@k8s-bot node e2e test this issue: #27296

**Comment by saad-ali** [comment]
LGTM. We can squash changes after LGTM from examples team.

Could some one from @kubernetes/examples take a look at this PR. 

In particular the auto-gen appears to be removing the `TAG RELEASE_LINK, added by the munger automatically` block from the docs, want to make sure that is expected.

**Comment by jeffmendoza** [comment]
Because the munger can't find the readme with the new path in the old release, it is not putting in the link (https://github.com/kubernetes/kubernetes/blob/master/cmd/mungedocs/unversioned_warning.go#L63).

Seems ok to me, the warning is still there that "You are looking at head"

**Comment by saad-ali** [comment]
Great. Cindy, I think we're good to squash these changes then.

**Comment by saad-ali** [comment]
With doc team singed-off, and changes squashed, adding LGTM.

**Comment by chrislovecnm** [comment]
We are having this problem now https://github.com/kubernetes/kubernetes/issues/27978 - did this change cause this problem??

**Comment by chrislovecnm** [comment]
Multiple e2e tests are failing with the old path.

**Comment by saad-ali** [comment]
Yes that does appear to be the case, looks like [some of the paths](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/examples.go#L494) in `test/e2e/examples.go` were not updated. Sounds like @dims has the fix ready, otherwise I can send one out.

**Pull request merged**